### PR TITLE
Document BUILDBOT_TEST_DB_URL + provide example of using DB from Docker

### PR DESCRIPTION
**Pull request by rutsky**

### CODE REVIEW

**Comment by rutsky** [comment]
TODO note at the end about not working MySQL 5.7 may be replaced with note that "MySQL 5.7 is currently unsupported, see issue #..." and appropriate issue on Buildbot bug tracker can be created.

File: master/docs/developer/database.rst [hunk 1]
```diff
@@ -1855,3 +1855,52 @@ id.
 
 If this weakness has a significant performance impact, it would be acceptable to
 conditionalize use of the subquery on the database dialect.
+
+Testing migrations with real databases
+--------------------------------------
+
+By default Buildbot test suite uses SQLite database for testings database
```
Comment (sa2ajj) [inline master/docs/developer/database.rst]: does not comply with one-sentence-per-line policy

**Comment by sa2ajj** [comment]
I think an issue and a link to it is the way to go.

File: master/docs/developer/database.rst (hunk 1 above)
Comment (rutsky) [inline master/docs/developer/database.rst]: I checked section above for that style policy --- it doesn't follow it either :)
It's a good policy IMO.

**Comment by rutsky** [comment]
I replaced TODO note with a link on http://trac.buildbot.net/ticket/3421.

File: master/docs/developer/database.rst (hunk 1 above)
Comment (sa2ajj) [inline master/docs/developer/database.rst]: We've been updating the docs to make it compliant.  Not all of it is though :(

**Comment by tardyp** [comment]
LGTM!

**Comment by rutsky** [comment]
@sa2ajj I resolved all stated issues, maybe merge?

File: master/docs/developer/database.rst [hunk 2]
```diff
@@ -1855,3 +1855,52 @@ id.
...
+   sudo docker run --name bb-test-mysql -e MYSQL_ROOT_PASSWORD=password \
+       -p 127.0.0.1:13306:3306 -d mysql:5.5
+   # Start interesting tests
+   BUILDBOT_TEST_DB_URL=mysql+mysqldb://root:password@127.0.0.1:13306/mysql \
+       trial buildbot.test
+
+.. note::
+
+   Currently Buildbot doesn't work with MySQL 5.7 Docker image, see issue
+   :bug:`3421`.
```
Comment (djmitche) [inline master/docs/developer/database.rst]: Yep, in fact BB's not compatible with 5.7 at all at the moment.

**Pull request merged**